use-timescale/migration/dual-write-and-backfill.md

```diff
@@ -0,0 +1,476 @@
+  Therefore, it's safe to remove it.
+
+Dump all plain tables and the TimescaleDB catalog from the source database:
+
+```
+pg_dump -d "$SOURCE" \
+  --format=plain \
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --exclude-table-data='_timescaledb_internal.*' \
+  --file=dump.sql
+```
+
+a. `--no-tablespaces` is required because Timescale does not support
+   tablespaces other than the default. This is a limitation.
+b. `--no-owner` is required because tsdbadmin is not a superuser and cannot
+   assign ownership in all cases. This flag means that everything will be
+   owned by the tsdbadmin user in the target regardless of ownership in the
+   source. This is a limitation.
+c. `--no-privileges` is required because tsdbadmin is not a superuser and
+   cannot assign privileges in all cases. This flag means that privileges
+   assigned to other users will need to be reassigned in the target
+   database as a manual clean-up task. This is a limitation.
+d. `--exclude-table-data='_timescaledb_internal.*'` will dump the structure
+   of the hypertable chunks, but not the data. This will create empty
+   chunks on the target ready for the backfill process.
+
+1. If the source database has the timescaledb extension installed in a schema
+   other than "public" it will cause issues on Timescale. Edit the dump file to
+   remove any references to the non-public schema. We need the extension in the
+   "public" schema on Timescale. This is a limitation.
+1. If any background jobs are owned by the "postgres" user, they need to be
+   owned by "tsdbadmin" on the target database. Edit the dump file accordingly.
+
+Load the roles and schema into the target database, and disable all background jobs.
+
+```
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  --echo-errors \
+  -f roles.sql \
+  -c 'select public.timescaledb_pre_restore();' \
+  -f dump.sql \
+  -f - <<'EOF'
+begin;
+select public.timescaledb_post_restore();
+
+-- disable all background jobs
+select public.alter_job(id::integer, scheduled=>false)
+from _timescaledb_config.bgw_job
+where id >= 1000
+;
+commit;
+EOF
+```
+
+<Highlight type="note">
+Background are disabled to prevent continuous aggregate refresh jobs from
+updating the continuous aggregate with incomplete/missing data. The continuous
+aggregates must be manually updated in the required range once the migration is
+complete.
+</Highlight>
+
+#### From plain PostgreSQL
+
+If you're migrating from plain PostgreSQL then we assume that you would like to
+convert some of your large tables which contain time-series data into
+hypertables. This step will consist of identifying those tables, excluding
+their data from the database dump, copying the database schema and tables, and
+setting up the time-series tables as hypertables. The data is backfilled into
+these hypertables in a subsequent step.
+
+Determine which tables to convert to hypertables:
```

Comment:
Maybe this is a good place for cross promoting https://github.com/timescale/doctor

Reply:
>I would prefer it if there was a docs page for timescale-doctor that we could point to though.
agreed. I wonder whether doctor is "ready to go" yet

Reply:
I reached out to Mats, who said:

> ... intention is to put it in the hands of users ... next steps are some cleaning and writing a blog